json/json.go

```diff
@@ -0,0 +1,88 @@
+// Package json provides a type and helpers for representing JSON log
+// lines.
+//
+// Format
+//
+// The JSON log line format consists of JSON objects, one per
+// \n-terminated line.
+//
+// Unilog recognizes three fields in that object that are considered
+// special (all are optional):
+//
+//    - ts: The time stamp of an event
+//    - canonical: Identifies the log event as "canonical", i.e. the
+//      most important line a service can log. It is considered to have
+//      the highest criticality level.
+//    - clevel: The criticality level of the event.
+//
+// Example
+//
+//    {"ts":"2006-01-02T15:04:05.999Z07:00","message":"hi there"}
+package json
+
+import (
+	"bytes"
+	"encoding/json"
+	"log"
+	"time"
+)
+
+// JSONLogLine is a representation of a generic log line that unilog
+// can destructure.
+type LogLine map[string]interface{}
+
+const defaultTimeFormat = time.RFC3339Nano
+
+// TS returns the timestamp of a log line; if a timestamp is set, TS
+// will attempt to parse it (first according to time.RFC3339Nano and
+// then time.RFC1123Z); if no timestamp is present, or the present
+// time stamp can not be parsed, TS returns the current time.
+func (j *LogLine) TS() time.Time {
```

Comment:
Is there a reason for the abbreviation `TS` instead of `Timestamp`? The latter is more self-evident when called. Or even `Time`, since it's not returning a Unix timestamp, but rather a parsed time field.

Reply:
My main reason was that it's a shorter name that we used in some of our documents before (9 vs. 2 chars saves quite a bit of bytes for lots of events), but it appears we already have log events that contain `"timestamp"` fields, so I would be tempted to make this smarter (and to definitely use "timestamp"!)

Reply:
To be clear, I was referring to the method name, not the field name that's written to disk. Though I guess it's arguably confusing to have the method name (in Go) differ from the field name that's eventually written.

Reply:
Okay, looking at the new version, since we now can have the field named either `ts` or `timestamp`, can we change this method name to `Timestamp`?